web/src/components/atoms/DownloadButton/index.tsx

```diff
@@ -5,37 +5,69 @@ import Button, { ButtonProps } from "@reearth-cms/components/atoms/Button";
 import Icon from "@reearth-cms/components/atoms/Icon";
 import { useT } from "@reearth-cms/i18n";
 
+export type PreviewType = "GEO" | "GEO3D" | "IMAGE" | "MODEL3D" | "UNKNOWN";
+
+export type AssetFile = {
+  children?: AssetFile[];
+  contentType?: string;
+  name: string;
+  path: string;
+  size: number;
+};
+
+export type Asset = {
+  id: string;
+  createdAt: string;
+  createdBy: string;
+  createdByType: string;
+  file: AssetFile;
+  fileName: string;
+  previewType?: PreviewType;
+  projectId: string;
+  size: number;
+  url: string;
+  threadId: string;
+  comments: Comment[];
+};
+
+export type Comment = {
+  id: string;
+  author: string;
+  content: string;
+  createdAt: string;
+};
+
 type DownloadButtonProps = {
   title?: string;
-  filename: string;
-  url: string;
+  selected?: Asset[];
```

Comment:
Can't we generalize that, and maybe accept other types, it's not mandatory to have an `asset` to make this atom work, right?
What do you think?

Reply:
Oh you are ahead of me. Like I mentioned in the reply above, making this component generic is definitely an option. Though, if we make it generic we will pull basically everything out of it that makes it unique to an Antd Button, no? 
Anyways, it is only being used with assets (schema and content pages). 
I think making it a common molecule as AssetDownloadButton or something is the best option.